atomicfu/src/jvmTest/kotlin/kotlinx/atomicfu/locks/NativeMutexTimeoutLincheckTest.kt

```diff
@@ -0,0 +1,75 @@
+package kotlinx.atomicfu.locks
+
+import org.jetbrains.kotlinx.lincheck.annotations.Operation
+import org.jetbrains.kotlinx.lincheck.annotations.Validate
+import org.jetbrains.kotlinx.lincheck.check
+import org.jetbrains.kotlinx.lincheck.strategy.managed.modelchecking.ModelCheckingOptions
+import org.jetbrains.kotlinx.lincheck.util.LoggingLevel
+import org.junit.Test
+import java.util.concurrent.ConcurrentHashMap
+import kotlin.time.Duration.Companion.nanoseconds
+
+class NativeMutexTimeoutLincheckTest {
+    class Counter {
+        @Volatile
```

Comment:
I wonder what this does. Since `Counter` is guarded by a lock, what purpose does `Volatile` serve?